akka-stream/src/main/scala/akka/stream/impl/fusing/Ops.scala

```diff
@@ -2285,6 +2285,281 @@ private[akka] final class StatefulMap[S, In, Out](
   override def toString = "StatefulMap"
 }
 
+/**
+ * INTERNAL API
+ */
+private[akka] final case class StatefulMapAsync[S, In, Out](parallelism: Int)(
+    create: () => Future[S],
+    //TODO (S, In) => (S, Future[Out]) seems more comfortable for user
+    f: (S, In) => Future[(S, Out)],
+    onComplete: S => Future[Option[Out]],
+    combineState: (S, S) => S)
+    extends GraphStage[FlowShape[In, Out]] {
+
+  import MapAsync._
+
+  private val in = Inlet[In]("StatefulMapAsync.in")
+  private val out = Outlet[Out]("StatefulMapAsync.out")
+
+  override def initialAttributes = DefaultAttributes.statefulMapAsync and SourceLocation.forLambda(f)
```

Comment:
Makes the `attributes` passed in as a parameter for later usage at `mapWithResourceAsync` and `foldResourceAsync`

Reply:
added.